lib/debug/thread_client.rb

```diff
@@ -659,18 +659,17 @@ def wait_next_action
 
     def wait_next_action_
       # assertions
-      raise "@mode is #{@mode}" unless @mode == :waiting
+      raise "@mode is #{@mode}" if !waiting?
```

Comment:
I think `if !state?` conveys the expectation clearer.

Reply:
`unless` is clear for me.